cli/src/commands/extensions/api.rs

```diff
@@ -110,11 +110,14 @@ async fn analyze(
     project: Option<String>,
     group: Option<String>,
 ) -> Result<ProjectId> {
-    // Ensure extension has file read-access.
-    let op_state = state.borrow();
-    let extension_state = op_state.borrow::<ExtensionState>();
-    extension_state.permissions.read.validate(&lockfile, "read")?;
-    drop(op_state);
+    // The block ensures that `op_state` is not held across an await. This could also be handled
+    // with `drop(op_state);`, but that leads to a false positive clippy lint (see rust-lang/rust-clippy#6353)
+    {
+        // Ensure extension has file read-access.
+        let op_state = state.borrow();
+        let extension_state = op_state.borrow::<ExtensionState>();
+        extension_state.permissions.read.validate(&lockfile, "read")?;
+    }
```

Comment:
It's likely cleaner to simply extract this code to a separate function (e.g., `check_permissions(op_state, &lockfile, "read")?;`), which would avoid the odd-looking block and the comment explaining why the odd-looking block is necessary.